### ISSUE TITLE
Transition out of ad using presentation time offset is broken between version 2.1 and 2.2

### ISSUE THREAD

**Issue by TheJohnBowers** [google/shaka-player#1237]
**Have you read the [FAQ](https://goo.gl/JE1Sy5) and checked for duplicate issues**:
Yes... I think this could be a duplicate of issue:  https://github.com/google/shaka-player/issues/1232, but I am not 100% sure
**What version of Shaka Player are you using**:

I build from branches so   
v2.2.x
v2.3.x

are broken

v2.1.x 
works fine

**Can you reproduce the issue with our latest release version**:
Yes
**Can you reproduce the issue with the latest code from `master`**:
Yes
**Are you using the demo app or your own custom app**:
Demo
**If custom app, can you reproduce the issue using our demo app**:
NA
**What browser and OS are you using**:
Chrome on macos
**What are the manifest and license server URIs**:
*(you can send the URIs to <shaka-player-issues@google.com> instead, but please use GitHub  

```js
{
    name: 'Segment list example with one ad',
    manifestUri: 'https://content.uplynk.com/d104c08c47b749159738ab81416fabbf.mpd?ck=1&ad=onenewad&srs=list', // gjslint: disable=110
    encoder: shakaAssets.Encoder.UPLYNK,
    source: shakaAssets.Source.UPLYNK,
    drm: [
      shakaAssets.KeySystem.WIDEVINE,
      shakaAssets.KeySystem.PLAYREADY,
      shakaAssets.KeySystem.CLEAR_KEY
    ],
    features: [
      shakaAssets.Feature.MP4,
      shakaAssets.Feature.PSSH,
      shakaAssets.Feature.MULTIKEY,
      shakaAssets.Feature.HIGH_DEFINITION
    ],
    licenseServers: {
      'com.microsoft.playready': 'https://content.uplynk.com/wv',
      'com.widevine.alpha': 'https://content.uplynk.com/wv',
      'org.w3.clearkey': 'https://content.uplynk.com/ck'
    },
    requestFilter: shakaAssets.UplynkRequestFilter,
    responseFilter: shakaAssets.UplynkResponseFilter
  },
  {
    name: 'Segment template example with one ad',
    manifestUri: 'https://content.uplynk.com/d104c08c47b749159738ab81416fabbf.mpd?ck=1&ad=onenewad', // gjslint: disable=110
    encoder: shakaAssets.Encoder.UPLYNK,
    source: shakaAssets.Source.UPLYNK,
    drm: [
      shakaAssets.KeySystem.WIDEVINE,
      shakaAssets.KeySystem.PLAYREADY,
      shakaAssets.KeySystem.CLEAR_KEY
    ],
    features: [
      shakaAssets.Feature.MP4,
      shakaAssets.Feature.PSSH,
      shakaAssets.Feature.MULTIKEY,
      shakaAssets.Feature.HIGH_DEFINITION
    ],
    licenseServers: {
      'com.microsoft.playready': 'https://content.uplynk.com/wv',
      'com.widevine.alpha': 'https://content.uplynk.com/wv',
      'org.w3.clearkey': 'https://content.uplynk.com/ck'
    },
    requestFilter: shakaAssets.UplynkRequestFilter,
    responseFilter: shakaAssets.UplynkResponseFilter
  },
```

**What did you do?**

So both of these assets are the same, one uses segment lists and one uses segment templates.  Transitioning from period 2 to 3 is broken.  About 2:42 in the asset.  2.2. and 2.3 seem to think there is a gap in the media, but things work fine with 2.1.x

**What did you expect to happen?**
see above - I expect it to play from period 2 to period 3

**What actually happened?**
Hang, and tell me that there is large gap

Ignoring large gap at 162.958886
Ignoring large gap at 165.804287

**Comment by TheModMaker** [comment]
I believe this was mostly fixed by 87c19da.  On `master` the first asset plays, but there is a problem with the second asset which is handled in #1232.

**Comment by TheJohnBowers** [comment]
So I tried both of these out on what is currently in the 2.2 and 2.3 branches.  In both cases the first now plays, but the second does not.  So If the cause of that is a duplicate of 1232 then you can close if you like.  If you want to notify me when 1232 has a fix I can retest this asset and be sure it is working.  However you want to handle it.

**Comment by TheModMaker** [comment]
#1232 has been fixed, can you please retest?

https://nightly-dot-shaka-player-demo.appspot.com/demo/

**Comment by TheJohnBowers** [comment]
I built it out of master and it appears to be working.
I built v2.3.x and the problem is still there.  I am assuming the fix is only in master.

**Comment by joeyparrish** [comment]
Yes, the fix for #1232 has been cherry-picked for v2.3.3. The correctness of the fix was still being debated while I was preparing v2.3.2. Sorry for any inconvenience caused by delaying this fix for one more release.

**Comment by joeyparrish** [comment]
I'm going to go ahead and close as a duplicate.  Thanks!
